power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -1060,45 +1075,54 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
         return sc_input;
     }
 
+    template <bool sym> void prepare_y_bus() {
+        std::vector<YBus<sym>>& y_bus_vec = get_y_bus<sym>();
+        // also get the vector of other Y_bus (sym -> asym, or asym -> sym)
+        std::vector<YBus<!sym>>& other_y_bus_vec = get_y_bus<!sym>();
+        // If no Ybus exists, build them
+        if (y_bus_vec.empty()) {
+            // check if other (sym/asym) Y_bus exists
+            bool const other_y_bus_exist = (!other_y_bus_vec.empty());
+            // assert(y_bus_vec.empty());
+            y_bus_vec.reserve(n_math_solvers_);
+            // get param, will be consumed
+            std::vector<MathModelParam<sym>> math_params = get_math_param<sym>();
+            // loop to build
+            for (Idx i = 0; i != n_math_solvers_; ++i) {
+                // if other y_bus exists, construct from existing Y_bus struct
```

Comment:
the code is quite self-explanatory, so maybe it's a good idea to remove the comments or replace them with fewer ones

```suggestion
        // If no Ybus exists, build them
        if (y_bus_vec.empty()) {
            bool const other_y_bus_exist = (!other_y_bus_vec.empty());
            y_bus_vec.reserve(n_math_solvers_);
            auto math_params = get_math_param<sym>();

            for (Idx i = 0; i != n_math_solvers_; ++i) {
                // construct from existing Y_bus structure if possible
```